homeassistant/components/ollama/config_flow.py

```diff
@@ -60,55 +61,51 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-STEP_USER_DATA_SCHEMA = vol.Schema(
-    {
-        vol.Required(CONF_URL): TextSelector(
-            TextSelectorConfig(type=TextSelectorType.URL)
-        ),
-    }
-)
-
-
 class OllamaConfigFlow(ConfigFlow, domain=DOMAIN):
     """Handle a config flow for Ollama."""
 
-    VERSION = 2
-    MINOR_VERSION = 2
+    VERSION = 3
+    MINOR_VERSION = 1
 
     def __init__(self) -> None:
         """Initialize config flow."""
         self.url: str | None = None
-        self.model: str | None = None
-        self.client: ollama.AsyncClient | None = None
-        self.download_task: asyncio.Task | None = None
+
+    @callback
+    def _get_user_schema(self, user_input: dict[str, Any] | None) -> vol.Schema:
+        """Return the schema for the user step."""
+        if user_input is None:
+            user_input = {}
+
+        return vol.Schema(
+            {
+                vol.Required(
+                    CONF_URL,
+                    description={"suggested_value": user_input.get(CONF_URL, "")},
+                ): TextSelector(TextSelectorConfig(type=TextSelectorType.URL)),
+            }
+        )
 
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
         """Handle the initial step."""
-        user_input = user_input or {}
-        self.url = user_input.get(CONF_URL, self.url)
-        self.model = user_input.get(CONF_MODEL, self.model)
-
-        if self.url is None:
+        if user_input is None:
             return self.async_show_form(
-                step_id="user", data_schema=STEP_USER_DATA_SCHEMA, last_step=False
+                step_id="user", data_schema=self._get_user_schema(user_input)
             )
 
         errors = {}
+        url = user_input[CONF_URL]
 
-        self._async_abort_entries_match({CONF_URL: self.url})
+        self._async_abort_entries_match({CONF_URL: url})
 
         try:
-            self.client = ollama.AsyncClient(
-                host=self.url, verify=get_default_context()
-            )
+            client = ollama.AsyncClient(host=cv.url(url), verify=get_default_context())
             async with asyncio.timeout(DEFAULT_TIMEOUT):
-                response = await self.client.list()
-
-            downloaded_models: set[str] = {
-                model_info["model"] for model_info in response.get("models", [])
-            }
+                await client.list()
+        except vol.Invalid:
+            errors["base"] = "invalid_url"
```

Comment:
Please don't mix up handling errors thrown by schema validation with errors thrown by the ollama client, we should separate in individual try-except clauses